state/execution.go

```diff
@@ -153,6 +153,11 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		panic(fmt.Sprintf("state machine returned an invalid prepare proposal response: expected last transaction to be a hash, got %d bytes", len(rpp.Txs[len(rpp.Txs)-2])))
 	}
 
+	rejectedTxs := len(rpp.Txs) - len(block.Txs)
```

Comment:
this number will likely be negative, as any rejected txs will not appear in the `rpp.Txs` but will appear in the `block.Txs`

also, we need to account for the last tx in `rpp.Txs`, which is a special hack to pass the data root back from the app.

Reply:
thanks for comment, addressed it